openlibrary/plugins/upstream/code.py

```diff
@@ -27,12 +27,13 @@
 import merge_authors
 
 if not config.get('coverstore_url'):
-    config.coverstore_url = "http://covers.openlibrary.org"
+    config.coverstore_url = "https://covers.openlibrary.org"
 
 class static(delegate.page):
     path = "/images/.*"
     def GET(self):
-        raise web.seeother('/static/' + web.ctx.path)
+        host = 'https://%s' % web.ctx.host if 'openlibrary.org' in web.ctx.host else ''
```

Comment:
This uses the gunicorn web.ctx var which under the hood appears to be http (resulting in insecure http request attempts for images)